toolchain/check/handle_impl.cpp

```diff
@@ -205,14 +205,12 @@ static auto PopImplIntroducerAndParamsAsNameComponent(
   auto [implicit_params_loc_id, implicit_param_patterns_id] =
       context.node_stack().PopWithNodeIdIf<Parse::NodeKind::ImplForall>();
 
-  ParameterBlocks parameter_blocks{
-      .implicit_params_id = SemIR::InstBlockId::Invalid,
-      .params_id = SemIR::InstBlockId::Invalid,
-      .return_slot_id = SemIR::InstId::Invalid};
   if (implicit_param_patterns_id) {
-    parameter_blocks =
+    auto parameter_blocks =
         CalleePatternMatch(context, *implicit_param_patterns_id,
                            SemIR::InstBlockId::Invalid, SemIR::InstId::Invalid);
+    CARBON_CHECK(parameter_blocks.call_params_id == SemIR::InstBlockId::Empty);
+    CARBON_CHECK(parameter_blocks.return_slot_id == SemIR::InstId::Invalid);
```

Comment:
Can you elaborate what's going on here? Previously, the result of `parameter_blocks.implicit_params_id` was retained. What's the effect of running it when the result is discarded?

Reply:
Thanks, what do you think of adding a comment here to explain the code? I've taken a stab above to try to clarify what I'm requesting.

Reply:
Done, with somewhat different wording.